src/transformers/data/data_collator.py

```diff
@@ -415,20 +414,17 @@ class DataCollatorForNextSentencePrediction:
     mlm_probability: float = 0.15
 
     def __call__(self, examples: List[Union[List[List[int]], Dict[str, torch.Tensor]]]) -> Dict[str, torch.Tensor]:
-        if isinstance(examples[0], (dict, BatchEncoding)):
```

Comment:
Those two lines need to be kept so the data collator works for inputs returned by a HF tokenizer.

Reply:
I have some confusion about this. Does `batch Encoding` always return a dict with `input_ids` as the key and need no more processing? Or follow the definition in the `TextDatasetForNSP`, return a dict with `tokens_a` and `tokens_b`?
Since the `create_features_from_example` method in the `DataCollatorForNextSentencePrediction` need  `tokens_a` and `tokens_b` for further processing. 
Does `input_ids` still need to be processed by `create_features_from_example`?

Reply:
Sry, there are bugs in the version I just submitted, please DO NOT merge them. thx

Reply:
Sorry I got confused. The inputs expected here are a list of dicts with some specific keys. Just document that properly and it should be good.

Reply:
I probably solved the problem mentioned before. The bugs in the code have also been fixed.

Reply:
It's just missing proper documentation of what this data collator now expects (so that users don't get confused if they don't use `TextDatasetForNSP`.